src/main/java/slimeknights/tconstruct/library/modifiers/Modifier.java

```diff
@@ -324,6 +330,41 @@ public void onInventoryTick(IModifierToolStack tool, int level, World world, Liv
   }
 
 
+  /**
+    * Called when this item is used when targeting a Block
+   * @param tool 
+   * @param level 
+   * @param stack 
+   * @param context        Full item use context
+   * @return  Action result
+   */
+  public ActionResultType onItemUse(ToolStack tool, int level, ItemStack stack, ItemUseContext context) {
```

Comment:
Back in 1.12, I had a interaction between rapiers and shields where the rapier would leap back, but return pass so the shield could block during the leap. Would that be worth supporting directly?

Its a bit weird behavior, I just ask now as supporting it in the API later would require using a different enum likely, which would be a breaking change.

Reply:
Why support directly? Nothing stops a modifier from doing their logic and returning a pass.

Reply:
And Shield and Leap can just check for each other.

Reply:
Mostly for a more general behavior. Saying rapier supports leaping back then any other tool action running. I suppose we could always do it via a pass return then returning success in another hook, so probably not a huge deal to support

Reply:
Yeah, just pass it.